hudi-common/src/main/java/org/apache/hudi/common/util/Option.java

```diff
@@ -98,6 +98,9 @@ public void ifPresent(Consumer<? super T> consumer) {
   }
 
   public <U> Option<U> map(Function<? super T, ? extends U> mapper) {
+    if (null == mapper) {
+      throw new NullPointerException("Expected a non-null value. Got null");
```

Comment:
let's make this more contextual?

`mapper is expected to be non-null`?

Reply:
Thanks.